HomeTask2.5/hash.js

```diff
@@ -0,0 +1,23 @@
+Object.prototype.hash = function (string) {
+    const strings = string.split(`.`);
+    const value = strings.reduce((obj, curString, i, arr) => {
+        if (!obj[curString]) {
+            arr.splice(i);//breaking the reduce once the obj[curString] becomes undefined
```

Comment:
This function is okay, try another way of writing this. May be recursion could help